.travis.yml

```diff
@@ -18,8 +18,9 @@ jobs:
       deploy:
           on:
             branch: master
-            tags: true
+            env:
+            - TRAVIS_COMMIT_MESSAGE = $TRAVIS_COMMIT_MESSAGE
           provider: script
-          skip_cleanup: true
+          skip_cleanup: true 
           script:
-            - npm run lerna:publish
+           - npm run lerna:deploy            
```

Comment:
Just thinking, would it be better to keep the logic for deployment and when to release in the travis config?

If an exact regexp to match semver could be laxed to the existence of a tag and the commit message, maybe this could work:
```yaml
jobs:
  include:
    - stage: release
      if: (tag =~ /^v/) AND (commit-message =~ /^chore\(release\)/)
      deploy:
        on:
          branch: master
        provider: script
        cleanup: true
        script: npm run lerna:publish
```

I think it would work with how you configured publishing:
```json
{
  "scripts": {
    "lerna:publish": "lerna publish from-package --yes",
  }
}
```

Reply:
Due to the fact that travis doesn't set the tag at deployment time unless you generate it at time, this won't work unfortunately
https://docs.travis-ci.com/user/deployment/releases/#when-tag-is-not-set-at-deployment-time